packages/api/src/controllers/experiment.ts

```diff
@@ -0,0 +1,215 @@
+import { Router } from "express";
+import _ from "lodash";
+import { v4 as uuid, validate as validateUuid } from "uuid";
+import { db } from "../store";
+import { NotFoundError, BadRequestError } from "../store/errors";
+import {
+  makeNextHREF,
+  parseFilters,
+  parseOrder,
+  toStringValues,
+} from "./helpers";
+import sql from "sql-template-strings";
+import { Experiment, ExperimentAudiencePayload, User } from "../schema/types";
+import { authorizer, validatePost } from "../middleware";
+import { WithID } from "../store/types";
+
+import experimentApis from "./experiment/index";
+import { ensureExperimentSubject } from "../store/experiment-table";
+
+async function toUserId(emailOrId: string) {
+  let user: User;
+  if (validateUuid(emailOrId)) {
+    user = await db.user.get(emailOrId);
+  } else {
+    const [users] = await db.user.find({ email: emailOrId });
+    user = users?.[0];
+  }
+  if (!user) {
+    throw new NotFoundError(`user not found: ${emailOrId}`);
+  }
+  return user.id;
+}
+
+const toUserIds = (emailsOrIds?: string[]) =>
+  Promise.all(emailsOrIds?.map(toUserId) ?? []);
+
+const app = Router();
+
+const experimentSubjectsOnly =
+  (experiment: string) => async (req, res, next) => {
+    await ensureExperimentSubject(experiment, req.user?.id);
+    return next();
+  };
+
+for (const [experiment, api] of Object.entries(experimentApis)) {
+  app.use(`/api/${experiment}`, experimentSubjectsOnly(experiment), api);
```

Comment:
I'm unhappy with the final paths of the experimental APIs. e.g.: `/api/experiment/api/lit-signing-condition/verify-lit-jwt`

I think the double `/api` looks weird but don't have many other ideas. Do you think `/x` would be too bad? i.e. `/api/experiment/x/lit-signing-condition/verify-lit-jwt`

Or otherwise I take any other ideas y'all might have!

Reply:
Hmm yeah, numbers would not be too bad but it'd seem pretty artificial to me, just to have some dummy value that we never update haha

Reply:
Hmmm what about an actual dummy `-`?
/api/experiment/-/lit-signing-condition/verify-lit-jwt

Reply:
Gone with `-` :)